piqe_ocp_lib/api/resources/ocp_cluster_versions.py

```diff
@@ -39,37 +44,163 @@ def update_cluster_version(self, cv_body):
         """
-        update_cluster_version_response = None
-        if cv_body:
-
-            # Add/Update resourceVersion under metadata and clusterID under spec to match of existing resource.These two
-            # field are also required to properly update ClusterVersion operator. Without these two fields,API call will
-            # succeed but changes won't be applied.
-            actual_cluster_version_response = self.get_cluster_version()
-            for cluster_version in actual_cluster_version_response.items:
-                if cluster_version["metadata"]["name"] == "version":
-                    cv_body["metadata"]["resourceVersion"] = cluster_version["metadata"]["resourceVersion"]
-                    cv_body["spec"]["clusterID"] = cluster_version["spec"]["clusterID"]
-
-            try:
-                update_cluster_version_response = self.ocp_cv.replace(body=cv_body)
-            except ApiException as e:
-                logger.exception("Exception while updating cluster version : %s\n" % e)
-        else:
+        if not isinstance(cv_body, dict):
             logger.error("Cluster version body is empty. Please provide config body with field to be updated")
+            return
+
+        # Add/Update resourceVersion under metadata and clusterID under spec to match of existing resource.These two
+        # field are also required to properly update ClusterVersion operator. Without these two fields,API call will
+        # succeed but changes won't be applied.
+        cluster_version = self.get_cluster_version()
+
+        cv_body["metadata"]["resourceVersion"] = cluster_version["metadata"]["resourceVersion"]
+        cv_body["spec"]["clusterID"] = cluster_version["spec"]["clusterID"]
 
-        return update_cluster_version_response
+        try:
+            return self.ocp_cv.replace(body=cv_body)
+        except ApiException as e:
+            logger.exception("Exception while updating cluster version : %s\n" % e)
 
-    def get_cluster_id(self):
+    def get_cluster_id(self) -> str:
         """
         Get cluster ID
-        :return: (str) cluster_id on success or None on failure
+        :return: cluster_id on success or None on failure
         """
-        cluster_id = None
         cluster_version_response = self.get_cluster_version()
-        for item in cluster_version_response.items:
-            if item["metadata"]["name"] == "version" and "clusterID" in dict(item["spec"]):
-                cluster_id = item["spec"]["clusterID"]
-        logger.info("Cluster ID : %s", cluster_id)
+        return cluster_version_response.spec.clusterID
+
+    def _build_spec(self, cv_body: Dict) -> Dict:
+        """
+        Add required fields in the user's input body.
+        :return: Updated dict with required values
+        """
+        cluster_version = self.get_cluster_version()
+
+        if not cluster_version:
+            return cv_body
```

Comment:
It looks like if we cannot obtain a cluster_version, we just return the provided cv_body as originally provided. In this case we should at least throwing out a warning and/or providing a useful message on what happened even if it's a private method.